src/foam/nanos/notification/EmailEnabledDAO.java

```diff
@@ -0,0 +1,102 @@
+package foam.nanos.notification;
+import foam.core.Detachable;
+import foam.dao.AbstractSink;
+import foam.nanos.notification.email.EmailMessage;
+import foam.core.FObject;
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+import foam.nanos.app.AppConfig;
+import foam.nanos.auth.User;
+import foam.nanos.logger.Logger;
+import foam.nanos.notification.email.EmailMessage;
+import foam.nanos.notification.email.EmailService;
+
+import java.text.NumberFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import static foam.mlang.MLang.AND;
+import static foam.mlang.MLang.EQ;
+
+public class EmailEnabledDAO extends ProxyDAO {
+
+  public EmailEnabledDAO(DAO delegate) {
+    setDelegate(delegate);
+  }
+
+  public EmailEnabledDAO(X x, DAO delegate) {
+    setX(x);
+    setDelegate(delegate);
+  }
+  @Override
+  public FObject put_(X x, FObject obj) {
+    DAO userDAO = (DAO) x.get("userDAO");
+    AppConfig config     = (AppConfig) x.get("appConfig");
+    EmailService email      = (EmailService) x.get("email");
+    ArrayList<String> users = new ArrayList<>();
+
+    Notification notif = (Notification) obj;
+    User user = (User) userDAO.find(notif.getUserId());
+
+
+    Notification oldNotif = (Notification) getDelegate().find(obj);
+
+    if ( oldNotif != null )
+      return super.put_(x, obj);
+
+    if ( ! notif.getEmailIsEnabled() )
+      return super.put_(x, obj);
+
+
+    if ( notif.getEmailName() == "notification" ) {
```

Comment:
Don't use == for comparing Strings in Java. Like this: if ( notif.getEmailName() == "notification" ) { ...
== compares that two objects are the exact same object, not the same value. So it can happen that "notification" == "notification" returns false.
Use "notification".equals(notif.getEmailName());
Or SafetyUtil.equals().
if ( notif.getEmailName() == "notification" ) will work sometimes, but if you get a call originating from the network which has built a new version of the String, it won't work.